fluentui_others/src/main/java/com/microsoft/fluentui/vNext/persona/AvatarView.kt

```diff
@@ -0,0 +1,355 @@
+package com.microsoft.fluentui.vNext.persona
+
+import android.content.Context
+import android.content.res.TypedArray
+import android.graphics.*
+import android.graphics.drawable.BitmapDrawable
+import android.graphics.drawable.Drawable
+import android.net.Uri
+import android.os.Build
+import android.provider.MediaStore
+import android.text.TextPaint
+import android.util.AttributeSet
+import androidx.appcompat.widget.AppCompatImageView
+import androidx.core.content.ContextCompat
+import com.microsoft.fluentui.R
+import com.microsoft.fluentui.generator.AvatarSize
+import com.microsoft.fluentui.generator.AvatarStyle
+import com.microsoft.fluentui.generator.resourceProxies.AvatarTokensSystem
+import com.microsoft.fluentui.persona.InitialsDrawable
+import com.microsoft.fluentui.util.FieldUpdateListener
+import com.microsoft.fluentui.util.ThemeUtil
+import java.lang.Math.pow
+import kotlin.math.sqrt
+
+
+/**
+ * [AvatarView] is a custom ImageView that displays the initials of a person on top of a colored circular
+ * background. The initials are extracted from their name or email. The color of the circular
+ * background is computed from the name and is based on an array of colors.
+ */
+open class AvatarView : AppCompatImageView {
+
+    var avatar: Avatar
+        set(value) {
+            field = value
+            setValues()
+        }
+
+    private var initialString: String = ""
+    private var internalPadding = 20
+    private var shouldUseCalculatedColors: Boolean = false
+    private var shouldUseDefaultImage: Boolean = false
+    private var hasImage = false
+    private var isImageSet = false
+    private var avatarBackgroundColor: Int = -1
+    private var avatarForegroundColor: Int = -1
+    private var avatarRingColor: Int = -1
+    private var avatarRingGapColor: Int = -1
+    private var presenceIconOutlineColor: Int = -1
+    private var avatarRingThickness: Float = 0f
+    private var avatarRingInnerThickness: Float = 0f
+    private var avatarRingOuterThickness: Float = 0f
+    private var presenceIconOutlineThickness: Float = 0f
+    private var presenceIconSize: Float = 0f
+    private val path: Path = Path()
+    private var fallbackBackground: Drawable? = null
+    private val tokenSystem: AvatarTokensSystem
+    private val textPaint: TextPaint = TextPaint(Paint.ANTI_ALIAS_FLAG).apply {
+        style = Paint.Style.FILL
+        density = context.resources.displayMetrics.density
+    }
+    private val paint: Paint = Paint(Paint.ANTI_ALIAS_FLAG)
+
+    @JvmOverloads
+    constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : super(context, attrs, defStyleAttr) {
```

Comment:
We need to wrap context with theme of others 
super(FluentUIContextThemeWrapper(appContext,R.style.Theme_FluentUI_Components)

Reply:
Similar change is required in constructor of PersonaListView.kt, please refer to the latest on develop branch. Without this, it may fail for an app not using Fluent Theme

Reply:
Cool, Thanks.
Yea i did it for ListView and Drawer but forgot it here. Thanks for pointing it out